rskj-core/src/test/java/co/rsk/peg/whitelist/WhitelistSupportImplTest.java

```diff
@@ -293,16 +311,58 @@ void verifyLockSenderIsWhitelisted_whenAddressIsNotWhitelisted_shouldReturnFalse
         assertFalse(actualResult);
     }
 
-    private void saveInMemoryStorageOneOffWhiteListEntry() {
-        Coin maxTransferValue = Coin.COIN;
-        OneOffWhiteListEntry oneOffWhiteListEntry = new OneOffWhiteListEntry(btcAddress, maxTransferValue);
-        List<OneOffWhiteListEntry> oneOffWhiteListEntries = Collections.singletonList(oneOffWhiteListEntry);
-        Pair<List<OneOffWhiteListEntry>, Integer> pairValue = Pair.of(oneOffWhiteListEntries, 100);
+    @Test
+    void save_whenLockWhitelistIsNull_shouldReturnZeroEntries() {
+        whitelistSupport.save();
 
-        inMemoryStorage.safeSaveToRepository(
-            LOCK_ONE_OFF.getKey(),
-            pairValue,
-            BridgeSerializationUtils::serializeOneOffLockWhitelist
-        );
+        int actualSize = whitelistSupport.getLockWhitelistSize();
+        assertEquals(0, actualSize);
+        assertNull(whitelistSupport.getLockWhitelistEntryByIndex(0));
+        assertNull(whitelistSupport.getLockWhitelistEntryByIndex(1));
+    }
+
+    @Test
+    void save_whenRSKIP87IsNotActive_shouldSaveOneOffLockWhitelistAddress() {
+        when(activationConfig.isActive(ConsensusRule.RSKIP87)).thenReturn(false);
```

Comment:
Let's use ActivationConfigForTest class as discussed

Reply:
Done! I uploaded it in a separate commit. Please ensure these are all the changes required.